javagen/src/main/java/com/azure/autorest/template/EnumTemplate.java

```diff
@@ -173,8 +218,27 @@ private void writeEnum(EnumType enumType, JavaFile javaFile, JavaSettings settin
                     enumBlock.annotation("JsonValue");
                 }
             }
-            enumBlock.publicMethod(typeName + " " + enumType.getToJsonMethodName() + "()",
+            enumBlock.publicMethod(typeName + " " + enumType.getToMethodName() + "()",
                 function -> function.methodReturn("this.value"));
+
+            if (settings.isStreamStyleSerialization()) {
+                String serializationCall = enumType.getElementType().jsonSerializationMethodCall("jsonWriter", null, "value");
+                enumBlock.annotation("Override");
+                enumBlock.publicMethod("JsonWriter toJson(JsonWriter jsonWriter) throws IOException",
+                    function -> function.methodReturn(serializationCall));
+
+                String deserializationCall = enumType.getElementType().jsonDeserializationMethod("jsonReader");
+                enumBlock.javadocComment(javadoc -> {
+                    javadoc.description("Reads a " + enumName + " from the JSON stream.");
+                    javadoc.line("<p>");
+                    javadoc.description("The passed JsonReader must be positioned at a " + enumType.getElementType().jsonToken() + " value.");
```

Comment:
This seems the only use of `jsonToken()` I found in PR.

Does it have other usages?

Reply:
This was the only usage I have for now. There may be more uses for this in the future but for now just this.